odl/discr/lp_discr.py

```diff
@@ -1248,6 +1267,275 @@ def discr_sequence_space(shape, exponent=2.0, impl='numpy', **kwargs):
+    each axis individually):
+
+    **0 arguments:**
+        Return a copy of ``discr``
+
+    **1 argument:**
+        [min,max]_corner -> keep sampling but translate domain so it
+        starts/ends at ``[min,max]_corner``
+
+        nsamples/cell_sides -> keep domain but change sampling.
+        See `uniform_partition` for restrictions.
+
+    **2 arguments:**
+        min_corner + max_corner -> translate and resample with the same
+        number of samples
+
+        [min,max]_corner + nsamples/cell_sides -> translate and resample
+
+        nsamples + cell_sides -> error due to ambiguity (keep
+        ``min_corner`` or ``max_corner``?)
+
+    **3+ arguments:**
+        The underlying partition is uniquely determined by the new
+        parameters. See `uniform_partition`.
+
+    See also
+    --------
+    uniform_partition : underlying domain partitioning scheme
+
+    Examples
+    --------
+    >>> discr = uniform_discr([0, 0], [1, 2], (10, 5))
+    >>> discr.cell_sides
+    array([ 0.1,  0.4])
+
+    No additional arguments -> copy:
+
+    >>> uniform_discr_fromdiscr(discr) == discr
+    True
+
+    1 or 2 arguments -> pick missing information from discr.
+
+    Changing ``min_corner`` or ``max_corner`` translates the domain.
+    Changing ``cell_sides`` or ``nsamples`` keeps the domain and
+    adapts the cell sides:
+
+    >>> uniform_discr_fromdiscr(discr, min_corner=[1, 1])
+    uniform_discr([1.0, 1.0], [2.0, 3.0], [10, 5])
+    >>> uniform_discr_fromdiscr(discr, max_corner=[0, 0])
+    uniform_discr([-1.0, -2.0], [0.0, 0.0], [10, 5])
+    >>> uniform_discr_fromdiscr(discr, cell_sides=[1, 1])
+    uniform_discr([0.0, 0.0], [1.0, 2.0], [1, 2])
+    >>> uniform_discr_fromdiscr(discr, nsamples=[5, 5])
+    uniform_discr([0.0, 0.0], [1.0, 2.0], [5, 5])
+    >>> uniform_discr_fromdiscr(discr, nsamples=[5, 5]).cell_sides
+    array([ 0.2,  0.4])
+
+    2 arguments -> pick missing information from discr.
+
+    Giving new ``min_corner`` and ``max_corner`` adapts the cell
+    sides, keeping the shape.
+    Combining one of ``min_corner`` and ``max_corner`` with ``nsamples``
+    computes the new domain by keeping ``cell_sides``.
+    Giving one of ``min_corner`` and ``max_corner`` together with
+    ``cell_sides`` keeps ``nsamples`` constant:
+
+    >>> uniform_discr_fromdiscr(discr, min_corner=[1, 1],
+    ...                                max_corner=[3, 3])
+    uniform_discr([1.0, 1.0], [3.0, 3.0], [10, 5])
+    >>> uniform_discr_fromdiscr(discr, min_corner=[1, 1],
+    ...                                max_corner=[3, 3]).cell_sides
+    >>> uniform_discr_fromdiscr(discr, min_corner=[1, 1],
+    ...                                nsamples=[5, 5])
+    uniform_discr([1.0, 1.0], [1.5, 3.0], [5, 5])
+    >>> uniform_discr_fromdiscr(discr, min_corner=[1, 1],
+    ...                                cell_sides=[1, 1])
+    uniform_discr([1.0, 1.0], [11.0, 6.0], [10, 5])
+
+    Specifying 3 or more new parameters will ignore ``discr``:
```

Comment:
But then only in the extreme case that this happens in all axes. If it's only in certain axes, I don't want to raise an error since consider it a valid use case.